s3backgrounddelete/s3backgrounddelete/object_recovery_msgbus.py

```diff
@@ -0,0 +1,192 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+"""Implementation of MessageBus for object recovery."""
+
+import json
+import socket
+import time
+import traceback
+from s3msgbus.cortx_s3_msgbus import S3CortxMsgBus
+from s3backgrounddelete.object_recovery_validator import ObjectRecoveryValidator
+
+class ObjectRecoveryMsgbus(object):
+
+    """This class is implementation of msgbus for object recovery."""
+
+    def __init__(self, config, logger):
+        """Initialize MessageBus."""
+        self._config = config
+        self._logger = logger
+        self.__msgbuslib = None
+        self.__isproducersetupcomplete = False
+        self._daemon_mode = config.get_daemon_mode()
+        self._sleep_time = config.get_msgbus_consumer_sleep_time()
+
+    def close(self):
+        """Closure and cleanup for ObjectRecoveryMsgbus."""
+        pass
+
+    def __process_msg(self, msg):
+        """Loads the json message and sends it to validation and processing."""
+        self._logger.info(
+            "Processing following records in consumer: " + msg)
+        try:
+            # msg: {"Key": "egZPBQAAAAA=-ZQIAAAAAJKc=",
+            #       "Value": "{\\"index_id\\":\\"egZPBQAAAHg=-YwIAAAAAJKc=\\",
+            #       \\"object_layout_id\\":1,
+            #        \\"object_metadata_path\\":\\"object1\\"}\\n"}
+            probable_delete_records = json.loads(msg)
+
+            if probable_delete_records:
+                validator = ObjectRecoveryValidator(
+                    self._config, probable_delete_records, self._logger)
+                validator.process_results()
+
+        except (KeyError, ValueError) as ex:    # Bad formatted message. Will discard it
+            self._logger.error("Failed to parse JSON data due to: " + str(ex))
+            return True
+        except Exception as ex:
+            self._logger.error(str(ex))
+            return False
+        return True
+
+    def __loadmsgbuslibrary(self):
+        """Load message bus library."""
+        #Basically loads msgbus lib and reads its config file
+        self._logger.debug("Instantiating S3MessageBus")
+        self.__msgbuslib = S3CortxMsgBus()
+        self._logger.debug("Connecting to S3MessageBus")
+        ret,msg = self.__msgbuslib.connect()
+        if not ret:
+            self._logger.error("msgbus connect failed {}".format(msg))
+            self.__msgbuslib = None
+        return ret
+
+    def receive_data(self,
+        consumer_id = None,
+        consumer_group = None,
+        msg_topic = None,
+        autoack = None,
```

Comment:
remove

Reply:
autoack removed. False Hardcoded in setup_receive